packages/runtime-common/module-syntax.ts

```diff
@@ -236,7 +236,15 @@ function makeNewField(
     suggestedCardName(fieldRef)
   );
 
-  return `@${fieldDecorator.name} ${fieldName} = ${fieldTypeIdentifier.name}(${fieldCardIdentifier.name});`;
+  if (
+    fieldRef.module.startsWith(baseRealm.url) &&
+    fieldRef.name === 'default'
+  ) {
+    // primitive fields
+    return `@${fieldDecorator.name} ${fieldName} = ${fieldTypeIdentifier.name}(${fieldCardIdentifier.name});`;
+  }
```

Comment:
I just noticed this, this is not a very robust way to determine if a card is primitive. keep in mind that a primitive could live in the local realm, or have be a named export. it's just a coincidence that all primitive fit this particular pattern right now. A better idea might be to run the card and check for the primitive symbol....

Reply:
> I just noticed this, this is not a very robust way to determine if a card is primitive. keep in mind that a primitive could live in the local realm, or have be a named export. it's just a coincidence that all primitives fit this particular pattern right now. A better idea might be to run the card and check for the primitive symbol....

I tried loading the card and checking for primitive symbol but the primitive symbol is only available in card-api and loading a card is async.

Reply:
I'm hoping that my dead lock fix will fix this problem too...